src/shims/posix/fs.rs

```diff
@@ -874,7 +916,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         let this = self.eval_context_mut();
 
         this.assert_target_os("linux", "statx");
-        this.check_no_isolation("`statx`")?;
+
+        // Reject if isolation is enabled.
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("`statx`", reject_with)?;
+            // statx never sets "EPERM". Set error code "ENOENT".
+            this.set_last_error_from_io_error(ErrorKind::NotFound)?;
```

Comment:
It looks like there is also `EACCES` -- would that be more accurate?

Reply:
Since `statx` supports both path name and fd, we'll have to set error code based on the input (pathname vs. fd) to be more accurate. So I moved this logic after inputs are resolved.

Reply:
> Since statx supports both path name and fd, we'll have to set error code based on the input (pathname vs. fd) to be more accurate.

I don't quite understand this comment...

Reply:
Oh I saw the new code, makes sense.